pkg/pool-weighted/contracts/WeightedMath.sol

```diff
@@ -223,6 +223,9 @@ library WeightedMath {
                 amountInWithoutFee = nonTaxableAmount.add(taxableAmount.sub(swapFee));
             } else {
                 amountInWithoutFee = amountsIn[i];
+                if (amountInWithoutFee == 0) {
```

Comment:
```suggestion
                amountInWithoutFee = amountsIn[i];
                
                // If a token's amount in is not being charged a swap fee then it might be zero (e.g. when joining a Pool with
                // only a subset of tokens). In this case, `balanceRatio` will equal `FixedPoint.ONE`, and the `invariantRatio`
                // will not change at all. We therefore skip to the next iteration, avoiding the costly `powDown` call.
                if (amountInWithoutFee == 0) {
```